### PR TITLE
Designate .jar file to parse

### PR DESCRIPTION
**Pull request by LibanOdowa**
Added a property in the .def file "j2objcJar" that can now be given a jar to parse.

Changed tests using test files on my computer to standalone ones.

### CODE REVIEW

File: Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/J2ObjCParser.kt [hunk 1]
```diff
@@ -15,7 +15,6 @@ import java.util.jar.JarFile
  *  @param classNode An ASM classNode
  */
 class J2ObjCParser(val classNode: ClassNode) {
-
```
Comment (jparachoniak) [inline Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/J2ObjCParser.kt:18]: revert this change?

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt [hunk 2]
```diff
@@ -247,6 +241,19 @@ private fun processCLib(flavorName: String, cinteropArguments: CInteropArguments
 
     val outKtPkg = fqParts.joinToString(".")
 
+    val jarDirectory = def.config.j2objcJar
```
Comment (jparachoniak) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt]: Jar files aren't often specified as a directory containing all jar files they are often specified as individual files ... but this may fit with the rest of the way headers work?

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt [hunk 3]
```diff
@@ -247,6 +241,19 @@ private fun processCLib(flavorName: String, cinteropArguments: CInteropArguments
...
+    if (language == Language.J2ObjC) {
+        val jarClassData = loadClassDataFromJar(JarFile(jarDirectory[0]))
+        val jarClassNodes = generateClassNodes(jarClassData)
+        j2objcIndexerResult =
+          IndexerResult(J2ObjCNativeIndex(jarClassNodes), CompilationWithPCH(emptyList<String>(), Language.J2ObjC))
+    } else {
+        j2objcIndexerResult =
+          IndexerResult(J2ObjCNativeIndex(emptyList<ObjCClass>()), CompilationWithPCH(emptyList<String>(), Language.J2ObjC))
+    }
+
```
Comment (jparachoniak) [inline Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt]: Why do this up here as opposed to as part of the if statement in indexResult?

val index:IndexerResult;
if (language == Language.J2ObjC) {
   ... this code ...
} else {
  nativeResult = buildNativeIndex(buildNativeLibrary(tool,def,cinteropArguments,imports), verbose)
}

or maybe pull this code out into a helper funtion and then we'd get

val index = if (language == Language.J2ObjC) buildJ2ObjCNativeIndex(buildNativeLibrary(tool,def,cinteropArguments,imports), verbose) else buildNativeIndex(buildNativeLibrary(tool,def,cinteropArguments,imports), verbose)

File: backend.native/tests/interop/j2objc/Foo.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+public class Foo {
```
Comment (jparachoniak) [inline backend.native/tests/interop/j2objc/Foo.java:1]: I think something needs to be changed in a build rule to force this to be compilied into a jar file too.

Comment (LibanOdowa) [reply backend.native/tests/interop/j2objc/Foo.java:1]: Yes, I agree. I kept it like this for now since I never ended up finding where the tests build-rules were originally. This is something that needs to be done though.

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt (hunk 3 above)
Comment (LibanOdowa) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt]: Yeah, I ended up changing my code to the first option ( I don't exactly know what I was thinking at the time, I was struggling with the language selection for some reason)

I do like the second idea more so I implemented that now.

File: Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/J2ObjCParser.kt (hunk 1 above)
Comment (LibanOdowa) [reply Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/J2ObjCParser.kt:18]: Done.

File: Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt (hunk 2 above)
Comment (LibanOdowa) [reply Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/jvm/main.kt]: Renamed to jarFiles

It's a list of space-seperated values that's in the files. So it would be just pointing to different files. Bad name in hindsight
